onnxruntime/wasm/api.h

```diff
@@ -27,15 +31,67 @@ extern "C" {
  * @param numThreads number of total threads to use.
  * @param logging_level default logging level.
  */
-void EMSCRIPTEN_KEEPALIVE OrtInit(int numThreads, int logging_level);
+int EMSCRIPTEN_KEEPALIVE OrtInit(int numThreads, int logging_level);
+
+/**
+ * create an instance of ORT session options.
+ */
+ort_session_options_handle_t EMSCRIPTEN_KEEPALIVE OrtCreateSessionOptions();
+
+/**
+ * release the specified ORT session options.
+ */
+void EMSCRIPTEN_KEEPALIVE OrtReleaseSessionOptions(ort_session_options_handle_t session_options);
+
+/**
+ * set an optimization level for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtSetSessionGraphOptimizationLevel(ort_session_options_handle_t session_options, size_t level);
+
+/**
+ * enable CPU memory arena for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtEnableCpuMemArena(ort_session_options_handle_t session_options);
+
+/**
+ * disable CPU memory arena for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtDisableCpuMemArena(ort_session_options_handle_t session_options);
+
+/**
+ * enable memory pattern for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtEnableMemPattern(ort_session_options_handle_t session_options);
+
+/**
+ * disable memory pattern for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtDisableMemPattern(ort_session_options_handle_t session_options);
+
+/**
+ * set an execution mode for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtSetSessionExecutionMode(ort_session_options_handle_t session_options, size_t mode);
+
+/**
+ * set a log ID for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtSetSessionLogId(ort_session_options_handle_t session_options, const char* logid);
+
+/**
+ * set a log severity level for session.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtSetSessionLogSeverityLevel(ort_session_options_handle_t session_options, size_t level);
 
 /**
  * create an instance of ORT session.
  * @param data a pointer to a buffer that contains the ONNX or ORT format model.
  * @param data_length the size of the buffer in bytes.
```

Comment:
add params for session_options

Reply:
I think it's a common practice not to put params when it's obvious from function name. But, I put comments that a handle must be freed by calling release function.

Reply:
yes that should be good enough